src/endian/big_endian.hpp

```diff
@@ -8,277 +8,185 @@
 #include <cassert>
 #include <cstdint>
 #include <cstring>
+#include <cmath>
 #include <limits>
 #include <type_traits>
 
 namespace endian
 {
 namespace detail
 {
-template<class ValueType, uint8_t Bytes>
-struct big
-{
-    static void put(ValueType value, uint8_t* buffer);
-    static void get(ValueType& value, const uint8_t* buffer);
-};
 
-template<class ValueType>
-struct big<ValueType, 1>
-{
-    static void put(ValueType value, uint8_t* buffer)
-    {
-        static_assert(sizeof(ValueType) <= 1, "ValueType too big.");
-        assert(buffer != nullptr);
-        *buffer = value;
-    }
+// Helper to convet floating point type into identically sized unsigned integer
+template<class Type>
+struct FloatType
+{ };
 
-    static void get(ValueType& value, const uint8_t* buffer)
-    {
-        static_assert(sizeof(ValueType) >= 1, "ValueType too small.");
-        assert(buffer != nullptr);
-        value = *buffer;
-    }
+template<>
+struct FloatType<float>
+{
+    static_assert(sizeof(float) == 4, "Float type must have a size of 4 bytes");
+    using UnsignedType = uint32_t;
 };
 
-template<class ValueType>
-struct big<ValueType, 2>
+template<>
+struct FloatType<double>
 {
-    static void put(ValueType value, uint8_t* buffer)
-    {
-        static_assert(sizeof(ValueType) <= 2, "ValueType too big.");
-        assert(buffer != nullptr);
-        buffer[1] = ((value >> 0) & 0xFF);
-        buffer[0] = ((value >> 8) & 0xFF);
-    }
-
-    static void get(ValueType& value, const uint8_t* buffer)
-    {
-        static_assert(sizeof(ValueType) >= 2, "ValueType too small.");
-        assert(buffer != nullptr);
-        value = (((ValueType) buffer[0]) << 8) |
-                (((ValueType) buffer[1]) << 0);
-    }
+    static_assert(sizeof(double) == 8, "Float type must have a size of 8 bytes");
+    using UnsignedType = uint64_t;
 };
 
-template<class ValueType>
-struct big<ValueType, 3>
+
+template<class ValueType, uint8_t Bytes>
+struct big_impl
 {
-    static void put(ValueType value, uint8_t* buffer)
+    static void put(ValueType& value, uint8_t* buffer)
```

Comment:
assert(buffer != nullptr);

Reply:
this check has already been made in the public API - in all the non-detail functions